data_architect_misc/data_transformer/transform_functions/apac_taiwan_transform_functions.py

```diff
@@ -0,0 +1,102 @@
+"""This is the subclass of Transform function for Taiwan (APAC division).
+
+We will define transform functions specific to Taiwan here.
+
+Author: Jholman Jaramillo
+Last Modified: September 10, 2020
+"""
+
+import re
+import os
+import pandas as pd
+from glob import glob
+
+import transform_errors
+from constants import comp_harm_constants
+from constants.transform_constants import KEY_CURRENT_INPUT_FILE, KEY_DELIMITER, KEY_HEADER
+from transform_functions.common_comp_harm_transform_functions import CommonCompHarmTransformFunctions
+from qa_functions.common_comp_harm_qa_functions import CommonCompHarmQAFunctions
+
+class ApacTaiwanTransformFunctions(CommonCompHarmTransformFunctions):
+    """
+    All custom (uncommon) transform functions **SPECIFIC to
+    individual processing task** must be defined as part
+    of this class.
+    """
+    TAIWAN_SPECIFIC_CATEGORY_MAPPINGS = {
+        "(?i).*Manual.*Powered.*TB.*": "Oral Care",
+
+        "(?i).*Air.*Fresheners.*": "Home Care",
+        "(?i).*Floor.*Cleaners.*": "Home Care",
+        "(?i).*Hand.*Dish.*": "Home Care",
+        "(?i).*Liquid.*Fabric.*": "Home Care",
+        "(?i).*Wipes.*Tissues.*": "Home Care",
+
+        "(?i).*Baby.*Accesories.*": "Personal Care",
+        "(?i).*Body.*Fragrance.*": "Personal Care",
+        "(?i).*Feminine.*Protection.*": "Personal Care",
+        "(?i).*Lipsticks.*Lip.*balm.*": "Personal Care",
+        "(?i).*Intimate.*Care.*": "Personal Care",
+        "(?i).*Regimen.*Products.*": "Personal Care",
+        "(?i).*Razor.*": "Personal Care",
+        "(?i).*Shaver.*": "Personal Care",
+        "(?i).*Soap.*": "Personal Care",
+        "(?i).*Talcum.*Powder.*": "Personal Care"
+    }
+
+    def __init__(self, config):
+        self.config = config
+
+    def create_new_dataframe_from_given_sheet_names_and_add_advertiser_or_category_column_using_sheet_name(
```

Comment:
Shouldn't this function go into common_comp_harm_transform_functions.py?

Reply:
This function is really specific for Taiwan as we talked before.